sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala

```diff
@@ -502,11 +502,31 @@ private[sql] object DataSourceV2Strategy {
 
   private def translateLeafNodeFilterV2(predicate: Expression): Option[Predicate] = {
     predicate match {
-      case PushablePredicate(expr) => Some(expr)
+      case PushablePredicate(expr) =>
+        if (expr.children().length == 2) {
+          expr.children()(0) match {
```

Comment:
could this be written into something like `expr.children() match { case (Literal, _) => `